server/stream.go

```diff
@@ -397,7 +397,16 @@ type stream struct {
 
 	monitorWg sync.WaitGroup // Wait group for the monitor routine.
 
-	batches *batching // Inflight batches prior to committing them.
+	// Inflight batches prior to committing them.
+	batches *batching
+
+	// State to check for batch completeness before applying it.
+	batchMu         sync.Mutex
```

Comment:
Should these all be in their own struct that we alloc only when we see first batch?

Reply:
Moved into a separate `batchApply` struct (https://github.com/nats-io/nats-server/pull/7067/commits/ef5c2f4dcfe9d4af29a9d95b9b1933d881b721d3).